### PR TITLE
Creates endpoints for Hasura mutations in resources

### PR DESCRIPTION
**Pull request by ypc-faros**
Creates endpoints for Hasura mutations in the resources directory. These are the endpoints used by the events CLI.

### CODE REVIEW

**Comment by cjwooo** [comment]
Please append `.gql` to all the endpoint file names to designate them as GraphQL files, and fix the spacing on some of the lines (some of them are indented with tabs instead of spaces, etc)

**Comment by cjwooo** [comment]
Does this still work if the query collection or mutation(s) already exist?

File: init/resources/hasura/endpoints/cicd_artifact [hunk 1]
```diff
@@ -0,0 +1,25 @@
+mutation insert_cicd_Artifact(
+$data_artifact_repository: String!,
+$data_artifact_id: String!,
+$data_artifact_organization: String!,
+$data_artifact_source: String!) {
+  insert_cicd_Artifact_one(object: {
+		uid: $data_artifact_id,
```
Comment (tovbinm) [inline init/resources/hasura/endpoints/cicd_artifact:7]: please use spaces for formatting

File: init/resources/hasura/endpoints/cicd_artifact [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        uid: $data_artifact_repository,
+        cicd_Organization: {
+          data: {
+            uid: $data_artifact_organization,
+            source: $data_artifact_source
+          },
+          on_conflict: {constraint: cicd_Organization_pkey, update_columns:[refreshedAt]}
+        }
+      },
+			on_conflict: {constraint: cicd_Repository_pkey, update_columns:[refreshedAt]}
```
Comment (tovbinm) [inline init/resources/hasura/endpoints/cicd_artifact:19]: same here - please use spaces for formatting

File: init/resources/hasura/endpoints/cicd_artifact [hunk 3]
```diff
@@ -0,0 +1,25 @@
+mutation insert_cicd_Artifact(
+$data_artifact_repository: String!,
```
Comment (tovbinm) [inline init/resources/hasura/endpoints/cicd_artifact:2]: does `String!` means nullable or not?

File: init/resources/hasura/endpoints/cicd_artifact_commit_association [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        uid: $data_artifact_id,
+        cicd_Repository: {
+          data: {
+            uid: $data_artifact_repository,
+            cicd_Organization: {
+              data: {
+                uid: $data_artifact_organization,
+                source: $data_artifact_source
+              },
+              on_conflict: {constraint: cicd_Organization_pkey, update_columns:[refreshedAt]}
```
Comment (tovbinm) [inline init/resources/hasura/endpoints/cicd_artifact_commit_association:23]: do we care about constraints? it's going to be a pain to maintain them in sql files and here.

File: init/resources/hasura/endpoints/cicd_build [hunk 5]
```diff
@@ -0,0 +1,23 @@
+mutation insert_cicd_Build($data_run_source: String!, $data_run_pipeline: String!, $data_run_organization: String!, $run_status: jsonb!, $data_run_id: String!) {
```
Comment (tovbinm) [inline init/resources/hasura/endpoints/cicd_build:1]: let's follow the same formatting as in other files

File: init/resources/hasura/endpoints/cicd_build_with_start_end [hunk 6]
```diff
@@ -0,0 +1,24 @@
+mutation insert_cicd_Build($run_start_time: timestamptz!, $data_run_source: String!, $data_run_pipeline: String!, $run_end_time: timestamptz!, $data_run_organization: String!, $run_status: jsonb!, $data_run_id: String!) {
```
Comment (tovbinm) [inline init/resources/hasura/endpoints/cicd_build_with_start_end:1]: same here - let's follow the same formatting as in other files

File: init/src/hasura/init.ts [hunk 7]
```diff
@@ -236,6 +236,50 @@ class HasuraInit {
...
+      type: 'create_query_collection',
+      args: {
+        name: "mutations",
+        definition: {
+          queries: mutations
+        }
+      },
+    });
+
+    await Promise.all(fs.readdirSync(directory).map(async file => {
```
Comment (tovbinm) [inline init/src/hasura/init.ts:265]: this fires all the requests in parallel. does the order matter?

Comment (cjwooo) [reply init/src/hasura/init.ts]: I don't think the order matters, @ypc-faros could you confirm?

File: init/resources/hasura/endpoints/cicd_artifact (hunk 3 above)
Comment (cjwooo) [reply init/resources/hasura/endpoints/cicd_artifact]: Not nullable

File: init/resources/hasura/endpoints/cicd_artifact_commit_association (hunk 4 above)
Comment (cjwooo) [reply init/resources/hasura/endpoints/cicd_artifact_commit_association]: Those constraints are necessary.

**Comment by cjwooo** [comment]
@ypc-faros Please run `npm run fix` to lint and format the new code. We haven't gotten around to including that in a commit hook yet.

File: init/src/hasura/init.ts (hunk 7 above)
Comment (ypc-faros) [reply init/src/hasura/init.ts]: Yep, order doesn't matter here because each promise is reading its own file.

**Review by tovbinm — COMMENTED** [review]
Can we start adding tests for these?

**Review by cjwooo — APPROVED** [review]

**Pull request merged**